packages/vow/src/watch-utils.js

```diff
@@ -79,20 +85,67 @@ export const prepareWatchUtils = (
     },
     {
       utils: {
+        /** @param {EVow<unknown>[]} vows */
+        all(vows) {
+          return this.facets.helper.createVow(vows, false);
+        },
+        /** @param {EVow<unknown>[]} vows */
+        allSettled(vows) {
+          return this.facets.helper.createVow(vows, true);
+        },
+        /** @type {AsPromiseFunction} */
+        asPromise(specimenP, ...watcherArgs) {
+          // Watch the specimen in case it is an ephemeral promise.
+          const vow = watch(specimenP, ...watcherArgs);
+          const promise = when(vow);
+          // Watch the ephemeral result promise to ensure that if its settlement is
+          // lost due to upgrade of this incarnation, we will at least cause an
+          // unhandled rejection in the new incarnation.
+          zone.watchPromise(promise, this.facets.retryRejectionPromiseWatcher);
+
+          return promise;
+        },
+      },
+      watcher: {
+        /**
+         * @param {unknown} value
+         * @param {object} ctx
+         * @param {bigint} ctx.id
+         * @param {number} ctx.index
+         * @param {number} ctx.numResults
+         * @param {boolean} ctx.isAllSettled
+         */
+        onFulfilled(value, ctx) {
+          this.facets.helper.processResult(value, ctx, 'fulfilled');
+        },
+        /**
+         * @param {unknown} reason
+         * @param {object} ctx
+         * @param {bigint} ctx.id
+         * @param {number} ctx.index
+         * @param {number} ctx.numResults
+         * @param {boolean} ctx.isAllSettled
+         */
+        onRejected(reason, ctx) {
+          this.facets.helper.processResult(reason, ctx, 'rejected');
+        },
+      },
+      helper: {
         /**
          * @param {EVow<unknown>[]} vows
```

Comment:
Pet peeve; the arguments are not vows, they're unknown specimens that might be vows.  That's what `EVow` describes:
```suggestion
        /**
         * @param {EVow<unknown>[]} specimens
```

Reply:
+1. regardless of pet peeves, this is a code interpretability issue worth improving.

incidentally, since `ERef` is "a ref or eventually a ref" one is likely to read `EVow` as "a vow or eventually a vow" but it's really "ERef or Vow".  We could consider some other names like `EVRef` or `VowRef`.

Reply:
Thinking about this further, the type should actually be:
```
* @param {unknown[]} specimens
```